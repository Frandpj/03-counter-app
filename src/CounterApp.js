import React, { useState } from 'react';
import PropTypes from 'prop-types';
// import React, { Fragment } from 'react';

// FC (Funtional components)

const CounterApp = ({ value = 10 }) => {

    const [counter, setCounter] = useState(value); // [] retorna un array


    // Handle
    const handleAdd = (e) => {
        // setCounter(counter + 1);
        setCounter((c) => c + 1);
    }
    const handleSubtract = (e) => {
        // setCounter(counter + 1);
        setCounter((c) => c - 1);
    }
    const handleReset = (e) => {
        // setCounter(counter + 1);
        setCounter((c) => value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>
            <button onClick={ handleAdd }>+1</button>
            <button onClick={ handleReset }>Reset</button>
            <button onClick={ handleSubtract }>-1</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number,
}

export default CounterApp;
