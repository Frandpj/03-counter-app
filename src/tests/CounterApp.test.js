import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';

import CounterApp from "../CounterApp";

describe('Pruebas de <CounterApp />', () => {
    
    test('Debe de mostrar <CounterApp /> correctamente', () => {
   
        const wrapper = shallow(<CounterApp />);

        // Crea un snapshot de lo que enviamos, en este caso el componente completo
        // <CounterApp />
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar el valor por defecto de 100', () => {
        
        const wrapper = shallow(<CounterApp value = {100} />);

        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('100');

    })
    
    

});
