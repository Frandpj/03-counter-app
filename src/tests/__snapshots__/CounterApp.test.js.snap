// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <CounterApp /> Debe de mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
